infer/src/integration/JsonReports.ml

```diff
@@ -241,6 +353,132 @@ module IssuesJson = struct
+    let should_report_proc_name =
+      Config.debug_mode || Config.debug_exceptions || not (BiabductionModels.mem proc_name)
+    in
+    if
+      error_filter source_file err_key.issue_type
+      && should_report_proc_name
+      && should_report err_key.issue_type err_key.err_desc
+    then
+      let message_content =
+        let base_qualifier = error_desc_to_plain_string err_key.err_desc in
+        if IssueType.(equal resource_leak) err_key.issue_type then
+          match Errlog.compute_local_exception_line err_data.loc_trace with
+          | None ->
+              base_qualifier
+          | Some line ->
+              let potential_exception_message =
+                Format.asprintf "%a: %s %d" MarkupFormatter.pp_bold "Note"
+                  potential_exception_message line
+              in
+              Format.sprintf "%s@\n%s" base_qualifier potential_exception_message
+        else base_qualifier
+      in
+      let message = 
+        {Sarifbug_j.text= message_content}
+      in
+      let level = String.lowercase (IssueType.string_of_severity err_key.severity) in
+      let ruleId = err_key.issue_type.unique_id in
+      let file =
+        SourceFile.to_string ~force_relative:Config.report_force_relative_path source_file
+      in
+      let file_loc = {Sarifbug_j.uri= file} in
+      let column = err_data.loc.Location.col in
+      let region = 
+        match column with
+        | -1 ->
+          { Sarifbug_j.startLine= err_data.loc.Location.line
+          ; startColumn= 1 }
+        | _ ->
+          { Sarifbug_j.startLine= err_data.loc.Location.line
+          ; startColumn= column }
+      in
+      let physical_location = 
+        {Sarifbug_j.artifactLocation= file_loc
+        ; region } 
+      in
+      let file_location_to_record = [{ Sarifbug_j.physicalLocation=physical_location}] in
+      let thread_flow_locs = 
+        [{Sarifbug_j.locations= loc_trace_to_sarifbug_record err_data.loc_trace err_key.severity}]
+      in
+      let thread_flow = 
+        [{Sarifbug_j.threadFlows= thread_flow_locs}]
+      in
+      let result =
+        { Sarifbug_j.message
+        ; level
+        ; ruleId 
+        ; codeFlows= thread_flow
+        ; locations= file_location_to_record }
+      in
+      Some (Sarifbug_j.string_of_sarifbug result)
+    else None
+end)
+
+module IssuesSarif = struct
+  include SarifIssuePrinter
+
+  (** Write bug report in SARIF 2.1 format *)
+  let pp_issues_of_error_log fmt error_filter _ proc_loc_opt proc_name err_log =
+    Errlog.iter
+      (fun err_key err_data -> pp fmt {error_filter; proc_name; proc_loc_opt; err_key; err_data})
+      err_log
+  
+  (** Write bug rules in SARIF 2.1 format *)
+  let pp_rules_of_error_log fmt error_filter _ proc_loc_opt proc_name err_log =
+    Errlog.iter
+      (fun err_key err_data -> pp_rules fmt {error_filter; proc_name; proc_loc_opt; err_key; err_data})
+      err_log
```

Comment:
Instead of starting from `Errlog`, would it be possible to start from report.json? The json report should be the source of truth as it has some additional filtering and tweaking. Starting from the json will avoid having duplicate code between the json output and the sarif output, eg `should_report_proc_name`, the special case for resource leaks, etc. Other reports emitted also start from the json output for the same reasons, see for example `TextReport` and `XMLReport`.